internal/pkg/util/bin/bin.go

```diff
@@ -23,28 +23,54 @@ import (
 )
 
 // FindBin returns the path to the named binary, or an error if it is not found.
+// We don't list any default because we want a deliberate decision about whether
+// to use the SuidBinaryPath which is more restrictive when in the suid flow.
 func FindBin(name string) (path string, err error) {
 	switch name {
 	// Basic system executables that we assume are always on PATH
-	case "true", "mkfs.ext3", "cp", "rm", "dd":
-		return findOnPath(name)
-	// Bootstrap related executables that we assume are on PATH
-	case "mount", "mknod", "debootstrap", "pacstrap", "dnf", "yum", "rpm", "curl", "uname", "zypper", "SUSEConnect", "rpmkeys", "squashfuse", "fuse-overlayfs", "fakeroot":
-		return findOnPath(name)
-	// Configurable executables that can be overridden in
-	// apptainer.conf. If config value is "" will look on PATH.
-	case "unsquashfs", "mksquashfs", "go", "cryptsetup", "ldconfig", "nvidia-container-cli":
-		return findFromConfigOrPath(name)
-	// distro provided setUID executables that are used in the fakeroot flow to setup subuid/subgid mappings
-	case "newuidmap", "newgidmap":
-		return findOnPath(name)
+	// We will search for these only in default PATH when in the suid flow
+	case "cp",
+		"dd",
+		"mkfs.ext3",
+		"mknod",
+		"mount",
+		"rm",
+		"true",
+		"uname":
+		return findOnPath(name, true)
+	// Executables that might be run privileged from the suid flow
+	// We must not search the user's PATH when in the suid flow with these
+	case "cryptsetup":
```

Comment:
Could be moved with others above

Reply:
Yes but I like it separate to emphasize its importance.